tests/solvers/test_base_solver.py

```diff
@@ -0,0 +1,62 @@
+from typing import Tuple
+
+import pytest
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from tests._utils import ATOL, RTOL, MockSolverOutput
+
+
+class TestBaseSolverOutput:
+    @pytest.mark.parametrize("batch_size", [1, 4])
+    @pytest.mark.parametrize("threshold", [0.0, 1e-1, 1.0])
+    @pytest.mark.parametrize("shape", [(7, 2), (91, 103)])
+    def test_sparsify_threshold(self, batch_size: int, threshold: float, shape: Tuple[int, int]) -> None:
+        rng = np.random.RandomState(42)
+        tmap = np.abs(rng.rand(shape[0], shape[1]))
+        output = MockSolverOutput(tmap)
+        output.sparsify(mode="threshold", threshold=threshold, batch_size=batch_size)
+        res = output.sparse_transport_matrix
+        assert isinstance(res, csr_matrix)
```

Comment:
use `import scipy.sparse as sp` and `assert sp.isspmatrix_csr`

Reply:
Actually, these checks aren't really necessary, would remove.

Reply:
Why are they not necessary? too obvious you mean?

Reply:
Yes, from the return type - you expect this is honored by how we specified the types so no need to check this.

Reply:
btw just needed it when adapting the code as I initially transposed the `csr_matrix` to arrive at a `csc_matrix` (not doing this any more as I transpose before conversion to a `csr_matrix`.